plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiSmokeTest.java

```diff
@@ -544,6 +545,32 @@ public void testHudiLongTimestampType()
         testTimestampMicros(HiveTimestampPrecision.NANOSECONDS, LocalDateTime.parse("2020-10-12T16:26:02.906668"));
     }
 
+    @Test
+    public void testHudiCustomKeygenEpochMillisPartitionedTables()
+    {
+        // NOTE: As of now, the partition_path value that is synced to metastore will be returned instead of the raw value that is used by the keygen
+        Session session = SessionBuilder.from(getSession()).build();
+        @Language("SQL") String actualQuery = "SELECT _hoodie_partition_path, part_country, part_date FROM " + HUDI_CUSTOM_KEYGEN_PT_V8_MOR;
+        @Language("SQL") String expectedQuery = "VALUES ('part_country=MY/part_date=2025-05-13', 'MY', '2025-05-13')," +
+                "('part_country=CN/part_date=2025-06-05', 'CN', '2025-06-05')," +
+                "('part_country=US/part_date=2025-06-06', 'US', '2025-06-06')," +
+                "('part_country=SG/part_date=2025-06-06', 'SG', '2025-06-06')," +
+                "('part_country=SG/part_date=2025-06-06', 'SG', '2025-06-06')," +
+                "('part_country=SG/part_date=2025-06-07', 'SG', '2025-06-07')," +
+                "('part_country=SG/part_date=2025-06-07', 'SG', '2025-06-07')";
+        assertQuery(session, actualQuery, expectedQuery);
+
+        // Ensure that partition pruning is working (using partition_path value) of level 3 partition_path value
+        @Language("SQL") String actualPartPruningQuery = actualQuery + " WHERE part_date='2025-06-06'";
```

Comment:
```suggestion
        @Language("SQL") String actualPartitionPruningQuery = actualQuery + " WHERE part_date='2025-06-06'";
```

Reply:
Let's follow Trino's development guildline to reduce possible review comments during upstreaming
https://github.com/trinodb/trino/wiki/Trino-Development-Guidelines#formatting
- Avoid abbreviations or acronyms in names. These generally make code harder to read as they are both technical domain specific and language specific. For example, use executorService or simply executor rather than execSvc. There are a few exceptions to this, such as stats vs statistics. Use the most concise name that represents the concept. See Clean Code chapter 2.

Reply:
Okay! Noted!